USAGOV 502 all topics

IsabelLaurenceau opened the pull request:
<!--- Provide a general summary of your changes in the title above -->
## Jira Task
<!--- Provide a link to the Jira ticket -->
https://cm-jira.usa.gov/browse/USAGOV-502 
(see also USAGOV-500)

## Description
<!--- Describe your changes in detail --> copying from docs/homepage.md 
# To get the homepage working
1. Add the taxonomy term "Home Page" by:
    1. Going to Structure -> Taxonomy -> Edit page type --> Add new term -> "Home Page"
Taxonomy terms are content in drupal and if not already in database need to be manually added.

2. Import config changes by running "drush cr"
3. Go to the homepage and press edit to make sure that the config changes are imported into drupal

    * If they are you should see homepage banner and homepage top links in basic page cms

    * If they are not you will not see a Home Page Banner and/or Top Links

4. On Homepage in Drupal add
    1. Body text for the welcome box,
        * We are here to help you navigate your Federal Government. Check out our Top Pages section below to see what others are looking for, or jump to All Topics and Services.
    2. "Home Page" as the page type,
        * See above if not an option
    3. A homepage banner
        * Any photo you'd like for testing
    4. Top links
        * Prepare for hurricane season,
        * Check the status of my tax refund,
        * Reach out to a Federal Agency,
        * Get government benefits and financial assistance

**Note** --> If homepage banner does not show after doing the above, refresh the page. The image styles may not have loaded. Similarly if you change the size of your window you may need to refresh. This should not be an issue on dev, stage or prod as tome will create the necessary files for the static site. If however you still do not see the banner go to content -> media -> media library. Try and upload media there. If it works there is a bug somewhere else in the code. If it does not there is an issue with media in general on the site and should be fixed by creating a new ticket for it.

## Checklist for the Developer
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask for help! -->
- [x] A link to the JIRA ticket has been added above.
- [x] The JIRA ticket identifies the desired result of this change.
- [x] The JIRA ticket contains clear acceptance criteria.
- [x] The JIRA ticket contains clear testing steps.
- [x] The JIRA ticket contains a description of "Done".
- [x] Any preparation/installation/update steps required for this code to work properly (drush commands, scripts, configuration updates) are documented in the ticket.

## Checklist for the Peer Reviewers
- [ ] The branch name of this PR matches the project standards.
- [ ] QA steps are followed and any changes to process are updated in the ticket.
- [ ] Code Standards are followed, there are no bad practices in use.
- [ ] Config changes (if any) include only necessary changes.
- [ ] No errors in Drupal or Client Browser.
- [ ] Code has been tested locally (if possible).
- [ ] Following AC and Testing Steps verify changes work as expected.
- [ ] There are no known side-effects outside of expected behavior.

IsabelLaurenceau committed:
USAGOV-500-homepage-hero: New Content Type for HomePage

IsabelLaurenceau committed:
Revert "USAGOV-500-homepage-hero: New Content Type for HomePage"

This reverts commit 8db2a36b0c48de4cba9954acb2db8c95ed221937.

IsabelLaurenceau committed:
USAGOV-500-homepage-hero: Hard-coded for jumping point

IsabelLaurenceau committed:
USAGOV-500-homepage-hero: hard coded for formatting

IsabelLaurenceau committed:
USAGOV-500-homepage-hero: Desktop todo: responsive for mobile & tablet

IsabelLaurenceau committed:
USAGOV-500-homepage-hero: Adding field(s) for top links for homepage

IsabelLaurenceau committed:
USAGOV-500-homepage-hero: Banner and message box are responsive README.md bin composer.json composer.lock config docker-compose.yml docs env.default env.local manifest-egress.yml manifest.yml modules patches phpunit.xml.dist s3 scripts sp.crt sp.key storage themes usagov.sql vendor web need to still make how do i box responsive

IsabelLaurenceau committed:
mobile-first-homepage: Different approach

IsabelLaurenceau committed:
mobile-first-homepage: How do I horizontal position

IsabelLaurenceau committed:
mobile-first-homepage: Three Issues and details left 1) Can't get banner to go to proper height 2) Can't get "How Do I box" to go to proper vertical position (-40/35 from message box) 3) Can't get "how do i buttons" to go full column

IsabelLaurenceau committed:
mobile-first-homepage: Details done just Three Issues 1) Can't get banner to go to proper height on resize 2) Can't get "How Do I box" to go to proper vertical position (-40/35 from message box) 3) Can't get "how do i buttons" to go full column

IsabelLaurenceau committed:
USAGOV-500-homepage-hero-banner: Homepage banner and message boxes

IsabelLaurenceau committed:
USAGOV-500-homepage-hero-banner: Small changes

IsabelLaurenceau committed:
USAGOV-500-homepage-hero-banner: Banner correct height on desktop

IsabelLaurenceau committed:
USAGOV-500-homepage-hero-banner: banner height for mobile (hard-coded)

IsabelLaurenceau committed:
life-events-homepage: start from the bottom blocks

IsabelLaurenceau committed:
life-events-homepage: Everything but banner mobile

IsabelLaurenceau committed:
USAGOV-501-life-events: Setting up the skeleton

IsabelLaurenceau committed:
USAGOV-501-life-events: skeleton

IsabelLaurenceau committed:
USAGOV-501-life-events: Skeleton Issues --> the how do i box is not in the flow so I can't get the life events and hero image to size correctly

IsabelLaurenceau committed:
less-grids: top layer on grid; background colors and banner to follow

IsabelLaurenceau committed:
less-grids: stopping for the night

IsabelLaurenceau committed:
less-grids: I think this is the cleanest way to do it Still need to do banner and backgrounds

IsabelLaurenceau committed:
less-grids: backgrounds

IsabelLaurenceau committed:
USAGOV-500-modular: width adjustments for backgrounds

IsabelLaurenceau committed:
USAGOV-500-modular: more width updates

IsabelLaurenceau committed:
USAGOV-500-modular: rename

IsabelLaurenceau committed:
USAGOV-500-modular: style changes

IsabelLaurenceau committed:
USAGOV-500-modular: chris's fix for blue background

IsabelLaurenceau committed:
USAGOV-500-modular: Blue banner extends width of viewport

IsabelLaurenceau committed:
USAGOV-500-modular: image goes full height

IsabelLaurenceau committed:
USAGOV-500-modular: all topics padding and margin

IsabelLaurenceau committed:
USAGOV-500-css-grid: css-grid implementation skeleton

IsabelLaurenceau committed:
USAGOV-500-css-grid: Update homepage.html.twig Styles in twig for faster dev will move once done; need to work on banner size and msg box;

IsabelLaurenceau committed:
USAGOV-500-css-grid: Got the banner to be responsive

IsabelLaurenceau committed:
USAGOV-500-css-grid: Message Box Formatting Works

IsabelLaurenceau committed:
USAGOV-500-css-grid: Banner reszes correctly

IsabelLaurenceau committed:
USAGOV-500-css-grid: Moved css into sass file

IsabelLaurenceau committed:
USAGOV-500-css-grid: Cleaned up grid for homepage

IsabelLaurenceau committed:
USAGOV-500-css-grid: border had a typo

IsabelLaurenceau committed:
USAGOV-502-all-topics: Test Node 723 for homepage

IsabelLaurenceau committed:
USAGOV-502-all-topics: Take the body and title from the node

IsabelLaurenceau committed:
USAGOV-502-all-topics: Life Events Top Portion Formatting and removing unnecessary files

IsabelLaurenceau committed:
USAGOV-502-all-topics: Moved styles into sass file

IsabelLaurenceau committed:
USAGOV-502-all-topics: Mobile changes for life-events header

IsabelLaurenceau committed:
USAGOV-502-all-topics: Most formatting of all topics done

IsabelLaurenceau committed:
USAGOV-502-all-topics: Just missing the icon

IsabelLaurenceau committed:
USAGOV-502-all-topics: Update views-view-list--sub-children.html.twig

IsabelLaurenceau committed:
USAGOV-502-all-topics: Icons

IsabelLaurenceau committed:
USAGOV-502-all-topics: mobile margins

IsabelLaurenceau committed:
USAGOV-502-all-topics: Update views-view-list--sub-children.html.twig

IsabelLaurenceau committed:
USAGOV-502-all-topics: config changes

IsabelLaurenceau committed:
USAGOV-502-all-topics: Get rid of extra padding

IsabelLaurenceau committed:
USAGOV-502-all-topics: Move welcome page to homepage

IsabelLaurenceau committed:
USAGOV-502-all-topics: Drupal config changes for homepage

IsabelLaurenceau committed:
USAGOV-502-all-topics: Added JUmp to all topics

IsabelLaurenceau committed:
USAGOV-502-all-topics: CSS fine-tuning

IsabelLaurenceau committed:
USAGOV-502-all-topics: Update node--1--full.html.twig

IsabelLaurenceau committed:
USAGOV-502-all-topics: CSS cleaning

IsabelLaurenceau committed:
USAGOV-502-all-topics: Make cards larger

IsabelLaurenceau committed:
USAGOV-502-all-topics: shifted the jump into their own rows for screen centering

IsabelLaurenceau committed:
USAGOV-502-all-topics: homepage

IsabelLaurenceau committed:
USAGOV-502-all-topics: Update _homepage.scss

IsabelLaurenceau committed:
USAGOV-502-all-topics: Update _homepage.scss

IsabelLaurenceau committed:
USAGOV-502-all-topics: Cards at all widths

IsabelLaurenceau committed:
USAGOV-502-all-topics: Requested changes

IsabelLaurenceau committed:
USAGOV-502-all-topics: Separating code out for readability and maintenance

IsabelLaurenceau committed:
USAGOV-502-all-topics: Move homepage cards

IsabelLaurenceau committed:
USAGOV-502-all-topics: Banner Image spans the width

IsabelLaurenceau committed:
USAGOV-502-all-topics: Update homepage-banner.html.twig

IsabelLaurenceau committed:
USAGOV-502-all-topics: Update _homepage.scss

IsabelLaurenceau committed:
USAGOV-502-all-topics: Hover color

IsabelLaurenceau committed:
USAGOV-502-all-topics: Update _homepage.scss

IsabelLaurenceau committed:
USAGOV-502-all-topics: Border color

IsabelLaurenceau committed:
USAGOV-502-all-topics: Mobile All Topics Header

IsabelLaurenceau committed:
USAGOV-502-all-topics: New red text border

IsabelLaurenceau committed:
USAGOV-502-all-topics: css variables and formatting

IsabelLaurenceau committed:
USAGOV-502-all-topics: style changes

IsabelLaurenceau committed:
USAGOV-500-responsive-banner: homepage banner from responsive image

IsabelLaurenceau committed:
USAGOV-500-responsive-banner: GOT IT

IsabelLaurenceau committed:
USAGOV-500-responsive-banner: got it

IsabelLaurenceau committed:
USAGOV-500-responsive-banner: Works as an image but not background

IsabelLaurenceau committed:
USAGOV-500-responsive-banner: got it with image-set

IsabelLaurenceau committed:
USAGOV-500-responsive-banner: background image but blurry

IsabelLaurenceau committed:
USAGOV-500-responsive-banner: Clunky but effective. Want to see if I can find a better way

IsabelLaurenceau committed:
USAGOV-500-responsive-banner: HomePage ReadMe

IsabelLaurenceau committed:
USAGOV-502-all-topics: Merge branch 'dev' into USAGOV-502-all-topics # Conflicts: # web/themes/custom/usagov/templates/views-view-list--sub-children.html.twig

IsabelLaurenceau committed:
USAGOV-502-all-topics: Update views-view-list--sub-children.html.twig

IsabelLaurenceau committed:
USAGOV-502-cards: flexbox solution

IsabelLaurenceau committed:
USAGOV-502-cards: Working correctly with grid

IsabelLaurenceau committed:
USAGOV-502-cards: Mobile Cards

IsabelLaurenceau committed:
USAGOV-502-all-topics: Banner config changes

IsabelLaurenceau committed:
USAGOV-502-all-topics: Needed changes for config to easily be sync'd

IsabelLaurenceau committed:
USAGOV-502-all-topics: Update page.html.twig

IsabelLaurenceau committed:
USAGOV-50x: Monday commit

IsabelLaurenceau committed:
USAGOV-502-all-topics: Refactored for mobile first

IsabelLaurenceau committed:
USAGOV-502-all-topics: Update newFooter.js

IsabelLaurenceau committed:
USAGOV-502-all-topics: Re-add hover effect for cards

IsabelLaurenceau committed:
Merge pull request #544 from usagov/USAGOV-50x

USAGOV-50x: Monday commit

IsabelLaurenceau committed:
USAGOV-502-all-topics: Merge carousel code in here

IsabelLaurenceau committed:
USAGOV-502-all-topics: tablet carousel

IsabelLaurenceau committed:
USAGOV-502-all-topics: Carousel

IsabelLaurenceau committed:
USAGOV-502-all-topics: Carousel View

IsabelLaurenceau committed:
USAGOV-502-all-topics: Carousel View Integrated

IsabelLaurenceau committed:
USAGOV-502-all-topics: Renaming of the view file

IsabelLaurenceau committed:
USAGOV-502-all-topics: New view config

IsabelLaurenceau committed:
USAGOV-502-all-topics: Life Events Width

IsabelLaurenceau committed:
USAGOV-502-all-topics: Cards

IsabelLaurenceau committed:
USAGOV-502-all-topics: small changes

IsabelLaurenceau committed:
USAGOV-502-all-topics: Implemented with usa-gov cards

IsabelLaurenceau committed:
USAGOV-502-all-topics: CSS changes

IsabelLaurenceau committed:
USAGOV-502-all-topics: Responsive Carousel

IsabelLaurenceau committed:
USAGOV-502-all-topics: Hide arrows

IsabelLaurenceau committed:
USAGOV-502-all-topics: moved css out of twig files

IsabelLaurenceau committed:
USAGOV-502-all-topics: style fixes

IsabelLaurenceau committed:
USAGOV-502-all-topics: Update _homepage.scss

IsabelLaurenceau committed:
USAGOV-502-all-topics: Update settings.local.php

IsabelLaurenceau committed:
USAGOV-502-all-topics: Merge branch 'dev' into USAGOV-502-all-topics # Conflicts: # web/themes/custom/usagov/templates/page.html.twig

IsabelLaurenceau committed:
USAGOV-502-all-topics: changed it to a button group

IsabelLaurenceau committed:
USAGOV-502-all-topics: Update _homepage.scss

IsabelLaurenceau committed:
Revert "USAGOV-502-all-topics: changed it to a button group"

This reverts commit e93cd041dac7fef9807daf9cbd933440c1590650.

IsabelLaurenceau committed:
USAGOV-502-all-topics: Update homepage-cards.html.twig

IsabelLaurenceau committed:
USAGOV-502-all-topics: Small changes

IsabelLaurenceau committed:
USAGOV-502-all-topics: style guide updates

IsabelLaurenceau committed:
USAGOV-502-all-topics: New Carousel Pagination

IsabelLaurenceau committed:
USAGOV-502-all-topics: Set the focus on new slide

IsabelLaurenceau committed:
USAGOV-502-all-topics: Focus for carousel

IsabelLaurenceau committed:
USAGOV-502-all-topics: buttons

IsabelLaurenceau committed:
USAGOV-502-all-topics: how do I are now link items

ednark committed:
USAGOV-592-media-upload: update libs

ednark committed:
USAGOV-592-media-upload: temp allow this branch to dev

ednark committed:
USAGOV-592-media-upload: temp allow this branch to dev

ednark committed:
USAGOV-592-media-upload: add s3fs patch

ednark committed:
USAGOV-592-media-upload: Merge branch 'dev' into USAGOV-592-media-upload

IsabelLaurenceau committed:
USAGOV-502-all-topics: homepage changes

ednark committed:
USAGOV-592-media-upload: set server HTTPS specifically in settings

ednark committed:
USAGOV-592-media-upload: Merge branch 'dev' into USAGOV-592-media-upload

IsabelLaurenceau committed:
USAGOV-502-all-topics: cleanup

IsabelLaurenceau committed:
USAGOV-502-all-topics: Merge branch 'USAGOV-592-media-upload' into USAGOV-502-all-topics

IsabelLaurenceau committed:
USAGOV-502-all-topics: Card alignment

IsabelLaurenceau committed:
USAGOV-502-all-topics: Change to svg for circles

IsabelLaurenceau committed:
USAGOV-502-all-topics: styling

IsabelLaurenceau committed:
USAGOV-502-all-topics: homepage banner responsive is back

IsabelLaurenceau committed:
USAGOV-502-all-topics: config changes for homepage banner

IsabelLaurenceau committed:
USAGOV-502-all-topics: Update HomePage.md

IsabelLaurenceau committed:
USAGOV-502-all-topics: Use short description in page view

IsabelLaurenceau committed:
USAGOV-502-all-topics: change block 5 to life_events_block

IsabelLaurenceau committed:
USAGOV-502-all-topics: thumbnails for carousel

IsabelLaurenceau committed:
USAGOV-502-all-topics: Jump image path was incorrect

akf commented:
I have a little feedback on the setup instructions. I probably did the right thing, because things look mostly right ...

This bit confused me a little. I think that's not meant to be two steps; since I don't fiddle with taxonomy much I found myself looking for a place to "add a term" before I went into the page type:  

> 1. Add the taxonomy term "Home Page"
> 2. Go to taxonomy -> edit page type --> add new term "Home Page"

I referred to your static site (I probably could have gone to Jira too) to get a little more detail on the body copy, since the markup seemed important. 

I did have to explicitly import the config changes. I'm sure that's because I spun up local dev with your code and then imported a database dump from production. If importing config hadn't worked, I think we'd want to look at why, rather than make the updates manually.

The homepage banner showed up right away after I added it.

akf commented:
A couple of things re: intros and short descriptions that I noticed on my local dev site: 

1. The carousel shows titles and images but no intro text. If I add a Short Description to one of the linked pages, that does show up. _I see now (from reading the Home Page doc) that this is intentional._ 
2. The cards in "All Topics and Services" use the Page Intro, even when I add a Short Description to the linked pages. I think they're supposed to use the short descriptions if they're available. 
3. The header over the "All topics" card is "All Topics and Services" and should be in sentence case ... which I see it is on your static site. Maybe that change is on a different branch? 

I'm just starting to look at the underlying structure.

akf commented:
I'm seeing this message in my Drupal recent log messages whenever I visit my home page: 

> User notice: Could not load image style max_1920w

I also see near-identical messages for "max_1280w" and "max_960w"

(There's more to the message, but my first question is -- do you see this too?)

akf commented:
I'm not sure whether you want to limit the scope of this review to USAGOV-502 and 500. I notice that the Life Events carousel is currently in alphabetical order, and I'll bet we want to be able to order those explicitly.

akf commented on web/themes/custom/usagov/templates/views-view-list--sub-children.html.twig:
```diff
@@ -20,75 +20,94 @@
 
-{% elseif term_name == "Standard Page" %}
-{# standard_page #}
+	{% if view.current_display == "block_4" %}
```
Would it be too late to change "block_4" to a friendlier name (like "Intro"?) Drupal has squirreled that away in the "Advanced" section of the block definition.

IsabelLaurenceau replied:
So these are what it was before I started on the homepage. I am not sure why they were named that way. You'll see for the life-events view I did rename it but I didn't want to touch the other blocks.

IsabelLaurenceau replied:
So these are what it was before I started on the homepage. I am not sure why they were named that way. You'll see for the life-events view I did rename it but I didn't want to touch the other blocks. 

I'm looking at the code again to make sure that it isn't used anywhere else but I should be able to rename it

IsabelLaurenceau replied:
Renamed!

akf commented on web/themes/custom/usagov/templates/views-view-list--sub-children.html.twig:
```diff
@@ -20,75 +20,94 @@
-      {% for row in rows %}
+{% if view.current_display == "block_4" and term_name != "Home Page"%}
+
+	{# block_4 is the intro_page display for this view which is used on the homepage #}
```
Am I correct in thinking that this clause is for backward compatibility and that we'll be able to remove it after the new Home Page has gone live?

IsabelLaurenceau replied:
> I'm seeing this message in my Drupal recent log messages whenever I visit my home page:
> 
> > User notice: Could not load image style max_1920w
> 
> I also see near-identical messages for "max_1280w" and "max_960w"
> 
> (There's more to the message, but my first question is -- do you see this too?)

I did but I hadn't seen that before. I had incorrectly named those 3 image styles. I pushed the name fixes up!

IsabelLaurenceau replied:
Yes and no. This did allow for the cards to show before going through the steps above but truly I just forgot to delete it before opening the PR. Deleted now.

akf commented on web/themes/custom/usagov/scripts/carousel.js:
```diff
@@ -0,0 +1,203 @@
+  carouselHeaders = document.querySelectorAll(".carouselHeaders");
+  makeDots();
+  slideDots = document.querySelectorAll(".navigation li button");
+  console.log("i did updates");
```
Let's remove or comment out these console.log statements before merging.

akf commented on web/themes/custom/usagov/scripts/carousel.js:
```diff
@@ -0,0 +1,203 @@
+        numSlides +
+        ": " +
+        carouselHeaders[i].textContent;
+      // li.setAttribute("aria-label", label );
```
Has a decision been made to omit the label? If yes, we can lose the lines above that construct it as well. :-) (If no, there may be some work to be done on those lines.)

IsabelLaurenceau replied:
No, that was actually supposed to remind me to put it back in when I switched to the svgs and it did not. But labels are now cleaned and reattached.

IsabelLaurenceau commented:
This does have me thinking though if I should include a fallback to use the page intro if short description is not set. 

-- I had forgotten that they also wanted the short description on the all topics cards because they didn't send those to me. I will edit that.

I will also edit the sentence case for the all topics and services.

IsabelLaurenceau commented:
> A couple of things re: intros and short descriptions that I noticed on my local dev site:
> 
> 1. The carousel shows titles and images but no intro text. If I add a Short Description to one of the linked pages, that does show up. _I see now (from reading the Home Page doc) that this is intentional._
> 2. The cards in "All Topics and Services" use the Page Intro, even when I add a Short Description to the linked pages. I think they're supposed to use the short descriptions if they're available.
> 3. The header over the "All topics" card is "All Topics and Services" and should be in sentence case ... which I see it is on your static site. Maybe that change is on a different branch?
> 
> I'm just starting to look at the underlying structure.

This does have me thinking though if I should include a fallback to use the page intro if short description is not set.

-- I had forgotten that they also wanted the short description on the all topics cards because they didn't send those to me. I will edit that.

I will also edit the sentence case for the all topics and services.

IsabelLaurenceau commented:
> I'm not sure whether you want to limit the scope of this review to USAGOV-502 and 500. I notice that the Life Events carousel is currently in alphabetical order, and I'll bet we want to be able to order those explicitly.

I agree, this should be for all the homepage tickets. 
Would it be content or us doing the reorder. We are using a view and I'm just trying to determine where it would be best to set that order. My initial thought is through the view ui on the cms

IsabelLaurenceau committed:
USAGOV-502-all-topics: Image style renaming

IsabelLaurenceau commented:
> I'm seeing this message in my Drupal recent log messages whenever I visit my home page:
> 
> > User notice: Could not load image style max_1920w
> 
> I also see near-identical messages for "max_1280w" and "max_960w"
> 
> (There's more to the message, but my first question is -- do you see this too?)

Yes! I didn't catch that before. I had incorrectly named those three image styles. Fix is pushed up.

IsabelLaurenceau committed:
USAGOV-502-all-topics: Update homepage-all-topics.html.twig

IsabelLaurenceau committed:
USAGOV-502-all-topics: short description with page intro fallback

IsabelLaurenceau committed:
USAGOV-502-all-topics: change block_4 to all_topics_block

IsabelLaurenceau committed:
USAGOV-502-all-topics: block layout for all_topics_block

IsabelLaurenceau committed:
USAGOV-502-all-topics: remove old cards

IsabelLaurenceau committed:
USAGOV-502-all-topics: Update HomePage.md

IsabelLaurenceau commented:
> I have a little feedback on the setup instructions. I probably did the right thing, because things look mostly right ...
> 
> This bit confused me a little. I think that's not meant to be two steps; since I don't fiddle with taxonomy much I found myself looking for a place to "add a term" before I went into the page type:
> 
> > 1. Add the taxonomy term "Home Page"
> > 2. Go to taxonomy -> edit page type --> add new term "Home Page"
> 
> I referred to your static site (I probably could have gone to Jira too) to get a little more detail on the body copy, since the markup seemed important.
> 
> I did have to explicitly import the config changes. I'm sure that's because I spun up local dev with your code and then imported a database dump from production. If importing config hadn't worked, I think we'd want to look at why, rather than make the updates manually.
> 
> The homepage banner showed up right away after I added it.

Updated the readMe (copied it to first comment above) Hopefully this is clearer!

IsabelLaurenceau committed:
USAGOV-502-all-topics: Update homepage-all-topics-jump.html.twig

IsabelLaurenceau committed:
USAGOV-502-all-topics: Merge branch 'dev' into USAGOV-502-all-topics

IsabelLaurenceau committed:
USAGOV-502-all-topics: Update views-view-field--sub-children.html.twig

IsabelLaurenceau committed:
USAGOV-502-all-topics: Update views-view-field--sub-children.html.twig

IsabelLaurenceau committed:
USAGOV-502-all-topics: Update responsive_image.styles.homepage_banner_sizing.yml

IsabelLaurenceau committed:
USAGOV-502-all-topics: Update homepage-how-box.html.twig

akf commented on composer.json:
```diff
@@ -157,6 +157,9 @@
             "drupal/views_menu_children_filter": {
                 "Use the technique from correctActiveTrail.patch to return the original menu item": "./patches/drupal/correctMenuChildren.patch"
+            },
+            "drupal/s3fs": {
```
Let's double-check with @ednark-gsa that these changes should go live before merging this. (Granted, it will be easy enough to reverse them.)

IsabelLaurenceau replied:
Sounds good! Thanks for looking at this!

akf reviewed (dismissed):
All looks good to me, although I want to be sure Dan's okay with the merged changes re: s3fs going to dev. (And I would like those console.log statements commented out or removed.) 

I'm checking Approve but not merging so you can get more eyes on this if you want.

IsabelLaurenceau committed:
USAGOV-502-all-topics: no footer gap and update to carousel.js logs

IsabelLaurenceau committed:
USAGOV-502-all-topics: Update carousel.js

IsabelLaurenceau committed:
USAGOV-502-all-topics: clean up the label

akf commented on web/themes/custom/usagov/scripts/carousel.js:
```diff
@@ -0,0 +1,189 @@
+    for (var i = 0; i < numSlides; i++) {
+      var li = document.createElement("li");
+      var pageNum = i + 1;
+      var label = `Card ${pageNum} of ${numSlides}: ${carouselHeaders[i].textContent.trim()}`;
```
Let's also guard against (remove) quotation marks in carouselHeaders. (I am wracking my brain for other things to worry about, but that's what I've got.)

IsabelLaurenceau committed:
USAGOV-502-all-topics: guard against quotes in string

IsabelLaurenceau committed:
USAGOV-502-all-topics: Update carousel.js

IsabelLaurenceau committed:
USAGOV-502-all-topics: homepage sass nesting cleanup

IsabelLaurenceau committed:
USAGOV-502-all-topics: Update _homepage.scss

ednark-gsa committed:
Merge branch 'dev' into USAGOV-502-all-topics

ednark committed:
USAGOV-502-all-topics: temp allow dev deploy

ednark committed:
USAGOV-502-all-topics: Merge branch 'USAGOV-502-all-topics' of github.com:usagov/usagov-2021 into USAGOV-502-all-topics

ednark committed:
USAGOV-502-all-topics: remove branch from temp deploy
